### PR TITLE
Show Services' Health Check Messages

### PR DESCRIPTION
**Pull request by danielsdeleo**
### :nut_and_bolt: Description: What code changed, and why?

Display services' health check messages in the applications tab UI. This enables expedited troubleshooting of problems when users emit helpful information from their health check scripts.

### :chains: Related Resources

#2554

NOTE: this doesn't include the show more/show less accordion for long messages. That will be done later.

### :athletic_shoe: How to Build and Test the Change

Build:
* `build components/automate-ui`

Test data:
There are some improvements to the sample data to show all of the different UI cases. Either restart the studio or `source .studio/applications-service`. Then `applications_populate_database`.

Verify Behavior:
* Filter for service name `pos-terminal`
* The four `pos-terminal.default` service groups show each of the following cases:
  * (production-us-west environment) no health check messages at all: shows informational message with link to documentation
  * (production-us-central environment)  stderr only: shows the stderr and no empty chrome for stdout
  * (production-us-mountain environment) stdout only: shows the stdout and no empty chrome for stderr
  * (production-us-east environment) stdout and stderr are both shown.
* filter for service name `nginx`
  * the bldr cache app in qa env shows messages of the maximum size. They wrap even without word breaks
  * the demo/demo one shows what we think a typical message might be.

### :camera: Screenshots, if applicable

Full page:
Screen Shot 2019-12-20 at 1 58 51 PM

Different health statuses (critical/warning/unknown/ok) with stdout, stderr, both, neither:
Screen Shot 2019-12-20 at 1 36 20 PM
Screen Shot 2019-12-20 at 1 36 34 PM
Screen Shot 2019-12-20 at 1 36 49 PM
Screen Shot 2019-12-20 at 2 00 58 PM

Screen Shot 2019-12-20 at 1 37 23 PM
Screen Shot 2019-12-20 at 1 37 45 PM
Screen Shot 2019-12-20 at 1 38 00 PM
Screen Shot 2019-12-20 at 1 38 11 PM
Screen Shot 2019-12-20 at 1 38 24 PM

### CODE REVIEW

**Comment by apriofrost** [comment]
Hi @danielsdeleo Overall this looks great!! I have some nitpick for you to consider iterating on:

1. **Text link**: Instead of having the entire sentence of "Learn more about writing health-check hooks" as a link to the doc, could we just have the "writing health-check hooks" with the link? This way, the text would match more closely with what the doc is about. It would look like this: image
2. **Radius size**: We talked about changing the radius size for all the new UI elements to be 4px. But looking at the result from the screenshots, I think the original 2px design would make it look sleeker and give higher priority to the content itself, rather than the shape of the container. I would suggest changing them all to 2px if it's possible or easy to do.
3. **Space between messages**: The space between the error and output messages when they co-exist seems a bit narrower than expected. In the image below, I heightened the areas where space should all be 8px.
image
4. **Missing divider**: In the mockup, there is also a divider between the service details info and the status-related notifiers. The distance from the divider to the element below and above it is 16px. 
image
Let me know if you have any questions about them. 😺

File: components/automate-ui/src/app/page-components/services-sidebar/services-sidebar.component.scss [hunk 1]
```diff
@@ -106,24 +107,121 @@ chef-alert {
     font-size: 14px;
   }
 
-  .health-line {
-    display: flex;
+  hr.health-info-divider {
+    margin: 16px -1em 16px -1em;
```
Comment (danielsdeleo) [inline components/automate-ui/src/app/page-components/services-sidebar/services-sidebar.component.scss]: So I think this semantically is more like a `hr` than a border on a div. But to do it that way means we have to negate the padding on the enclosing element which couples those two things 😬 🤷‍♀

File: components/automate-ui/src/app/page-components/services-sidebar/services-sidebar.component.scss [hunk 2]
```diff
@@ -106,24 +107,121 @@ chef-alert {
...
+    &.OK {
+      background-color: var(--chef-badge-success-background);
+      color: var(--chef-badge-success-color);
+    }
+
   }
 
-  .channel-line {
-    padding-left: 18px;
+  .disconnect-notice {
```
Comment (danielsdeleo) [inline components/automate-ui/src/app/page-components/services-sidebar/services-sidebar.component.scss]: Most of this isn't new, I moved it to be as nested as the other things so this file would be consistent with itself.

**Comment by danielsdeleo** [comment]
K, I made the following changes:

* make the `disconnect-notice` and `health-info-panel` and `no-healthcheck-messages` border radius 2px
* add horizontal rule between basic service info and health/connectedness info panels
* change the text that's in/out of the link tags
* set space between two `healthcheck-message`s to 8px

**Pull request merged**